python/cudf/cudf/bindings/cudf_cpp.pyx

```diff
@@ -44,7 +44,9 @@ gdf_dtypes = {
     GDF_INT16:             np.int16,
     GDF_INT8:              np.int8,
     GDF_BOOL8:             np.bool_,
+    GDF_DATE32:            np.datetime64,
     GDF_DATE64:            np.datetime64,
```

Comment:
Any reason we shouldn't just start using the TIMESTAMP types instead of the GDF_DATE64? I imagine it would simplify things in the backend.

Reply:
Initially I thought I was seeing an issue with pyarrow compat, but I just tested and am not seeing an issue anymore. I agree, using `pa.Timestamp(unit='ms')` instead of `pa.Date64` is more consistent.

I should note that since both the Arrow Timestamp[ms] and Date64 types are being converted to `np.datetime64[ms]` in our `from_arrow()` call, we lose the knowledge of the original Arrow DataType and will construct a different type when someone calls `to_arrow()`:

```python
import cudf
import pyarrow as pa

pa_array = pa.array([1, 2, 3], type=pa.date64())
cd_array = Series.from_arrow(pa_array).to_arrow()

print(type(pa_array)) # <class 'pyarrow.lib.Date64Array'>
print(type(cd_array)) # <class 'pyarrow.lib.TimestampArray'>
print(pa.Array.equals(pa_array, cd_array)) # False
```

Reply:
I think it's fine for us to always output a `timestamp` type here and ignore DATE64 for the most part.